### DISCUSSION TITLE
Shaders on buildings

### DISCUSSION THREAD

**Discussion by mogmog** [visgl/deck.gl#4473] [Q&A]
I have recently discovered the library Tangram which facilitates using maps and WebGL together, and they have some very pretty looking shaders on the sides of buildings - 

https://tangrams.github.io/tangram-sandbox/tangram.html?styles/tilt-ikeda#20/40.71430/-74.05934

All sorts of documentation is available here.

https://tangrams.readthedocs.io/en/master/Overviews/Shaders-Overview/

I am trying to work out if I can achieve something similar in Deck as I am fairly heavily invested in the Deck world and have more familarity with it.

Which Deck layers would be the right ones to us to work towards these kind of effects?

**Comment by tsherif** [comment]
Layer extensions allow you to inject custom code into a layer's shaders: https://deck.gl/#/documentation/developer-guide/writing-custom-layers/layer-extensions

If that isn't enough, you can also create fully custom layers (optionally subclassing existing layers) that give you (almost) full control over GL calls: https://deck.gl/#/documentation/developer-guide/writing-custom-layers/writing-your-own-layer

Extending or subclassing the PolygonLayer (which is used for the buildings in the [example](https://github.com/uber/deck.gl/blob/master/examples/website/trips/app.js) on the deck.gl site) could be a way to achieve similar effects.
